### PR TITLE
Add more tests for trace package.

### PR DESCRIPTION
**Pull request by euskadi31**

### CODE REVIEW

**Review by MadVikingGod — APPROVED** [review]

**Review by XSAM — APPROVED** [review]
Should we skip this changelog? These changes do not affect our end-users.

**Comment by euskadi31** [comment]
@XSAM done :)

**Review by XSAM — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

File: trace/trace_test.go [hunk 1]
```diff
@@ -62,6 +63,159 @@ func TestIsValid(t *testing.T) {
...
+		traceID: [16]byte{1},
+		spanID:  [8]byte{42},
+	}
+
+	if a.Equal(b) != true {
+		t.Error("Want: true, but have: false")
+	}
+}
+
+func TestIsSampled(t *testing.T) {
```
Comment (MrAlias) [inline trace/trace_test.go:82]:
```suggestion
func TestSpanContextIsSampled(t *testing.T) {
```

File: trace/trace_test.go [hunk 2]
```diff
@@ -62,6 +63,159 @@ func TestIsValid(t *testing.T) {
 	}
 }
 
+func TestEqual(t *testing.T) {
```
Comment (MrAlias) [inline trace/trace_test.go:66]:
```suggestion
func TestSpanContextEqual(t *testing.T) {
```

File: trace/trace_test.go [hunk 3]
```diff
@@ -62,6 +63,159 @@ func TestIsValid(t *testing.T) {
...
+	}
+
+	b := SpanContext{
+		traceID: [16]byte{1},
+		spanID:  [8]byte{42},
+	}
+
+	if a.Equal(b) != true {
+		t.Error("Want: true, but have: false")
+	}
```
Comment (MrAlias) [inline trace/trace_test.go:79]: Missing negative test where we verify equal returns false when it is expected.

File: trace/trace_test.go [hunk 4]
```diff
@@ -62,6 +63,159 @@ func TestIsValid(t *testing.T) {
...
+			have := sc.IsSampled()
+
+			if have != testcase.want {
+				t.Errorf("Want: %v, but have: %v", testcase.want, have)
+			}
+		})
+	}
+}
+
+func TestIsRemote(t *testing.T) {
```
Comment (MrAlias) [inline trace/trace_test.go:111]:
```suggestion
func TestSpanContextIsRemote(t *testing.T) {
```

File: trace/trace_test.go [hunk 5]
```diff
@@ -62,6 +63,159 @@ func TestIsValid(t *testing.T) {
...
+		tid  TraceID
+		sid  SpanID
+		want []byte
+	}{
+		{
+			name: "SpanContext.MarshalJSON() returns true if sc has both an Trace ID and Span ID",
+			tid:  [16]byte{1},
+			sid:  [8]byte{42},
+			want: []byte(`{"TraceID":"01000000000000000000000000000000","SpanID":"2a00000000000000","TraceFlags":"00","TraceState":"","Remote":false}`),
+		},
```
Comment (MrAlias) [inline trace/trace_test.go:152]: missing test cases verifying `TraceFlags`, `TraceState`, and `Remote` are correctly serialized.

File: trace/trace_test.go [hunk 6]
```diff
@@ -62,6 +63,159 @@ func TestIsValid(t *testing.T) {
...
+		}, {
+			name:  "Invalid SpanID with invalid char",
+			hex:   "80f198ee563433g7",
+			valid: false,
+		}, {
+			name:  "Invalid SpanID with uppercase",
+			hex:   "80f198ee53ba86F7",
+			valid: false,
+		}, {
+			name:  "Invalid SpanID with nil",
```
Comment (MrAlias) [inline trace/trace_test.go:196]:
```suggestion
			name:  "Invalid SpanID with zero value",
```

File: trace/trace_test.go [hunk 7]
```diff
@@ -62,6 +63,159 @@ func TestIsValid(t *testing.T) {
...
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			sid, err := SpanIDFromHex(testcase.hex)
+
+			if testcase.valid && err != nil {
+				t.Errorf("Expected SpanID %s to be valid but end with error %s", testcase.hex, err.Error())
+			}
+
+			if !testcase.valid && err == nil {
```
Comment (MrAlias) [inline trace/trace_test.go:208]:
```suggestion
			} else if !testcase.valid && err == nil {
```

File: trace/trace_test.go [hunk 8]
```diff
@@ -86,6 +240,10 @@ func TestIsValidFromHex(t *testing.T) {
 			name:  "Invalid TraceID with uppercase",
 			hex:   "80f198ee56343ba864fe8b2a57d3efF7",
 			valid: false,
+		}, {
+			name:  "Invalid TraceID with nil",
```
Comment (MrAlias) [inline trace/trace_test.go:244]:
```suggestion
			name:  "Invalid TraceID with zero value",
```

**Review by MrAlias — CHANGES_REQUESTED** [review]

File: trace/trace_test.go (hunk 3 above)
Comment (MrAlias) [reply trace/trace_test.go:79]: I'm not seeing where this was resolved. Can you point me in the right direction with a link to the added negative test?

Comment (euskadi31) [reply trace/trace_test.go:84]: Oops, an oversight on my part.

Comment (euskadi31) [reply trace/trace_test.go:84]: done

**Review by MrAlias — APPROVED** [review]

**Pull request merged**